### DISCUSSION TITLE
PRINT_END doesn't work

### DISCUSSION THREAD

**Discussion by 3D-Frank** [OrcaSlicer/OrcaSlicer#182] [General]
Hi!
I have now made the first test prints and I am very satisfied. Unfortunately, I have a hopefully solvable problem. After printing, the print head remains on the object instead of executing the Print_End command. The message in the console is: Unknown command:"PRINT_END".
I think I set everything up like in SuperSlicer and it works there.

Thank you for your support!

<img width="992" alt="grafik" src="https://user-images.githubusercontent.com/97818762/212560361-d95a41d6-2f6a-44a9-991d-655b9a4e8df1.png">

**Comment by erebus04** [comment]
@3D-Frank  The variables here are different than in Super Slicer.

[bed_temperature_initial_layer_single]
[nozzle_temperature_initial_layer]

This is from miy Klipper Maschine
![image](https://user-images.githubusercontent.com/34960970/212560931-d9dd9b8d-a93c-4f1b-9b4a-9763db0493f0.png)

**Reply by 3D-Frank** [reply]
Thanks! I can test that, but your end g-code is no different. My settings in the SuperSlicer are based on the input of the 3D CrydTeam and they are experts when it comes to clippers etc. As I said: In the SuperSlicer it works without any problems

**Reply by erebus04** [reply]
That may be, but the clipper Gcode for end is END_PRINT in the default

**Comment by SoftFever** [comment]
Can you attach your g-code?
As far as I can see from the gcode generated on my side and my test, there is no issue.
You may want to double-check what's the macro name you defined in you Klipper. 
![Screenshot 2023-01-16 at 9 24 24 PM](https://user-images.githubusercontent.com/103989404/212688599-ea9a1610-d1ff-45bb-b638-7944b3b3151d.png)

**Comment by 3D-Frank** [comment]
Hello and thanks for your help!
I added the "M117 Print End" G-code yesterday and that seems to be working. As I said: In the SuperSlicer it only works "Print_End". I am sending you two G-Code test files with and without "M117 Print End" and my Klipper Macro.cfg
[macros.txt](https://github.com/SoftFever/BambuStudio-SoftFever/files/10426971/macros.txt)
[Test_Gcode_only_Print_End.txt](https://github.com/SoftFever/BambuStudio-SoftFever/files/10426996/Test_Gcode_only_Print_End.txt)
[Test_Gcode_M117 End Print & Print_End.txt](https://github.com/SoftFever/BambuStudio-SoftFever/files/10426997/Test_Gcode_M117.End.Print.Print_End.txt)

**Comment by 3D-Frank** [comment]
Hello and thanks for your help!
I added the "M117 Print End" G-code yesterday and that seems to be working. As I said: In the SuperSlicer it only works "Print_End". I am sending you two G-Code test files with and without "M117 Print End" and my Klipper Macro.cfg
[macros.txt](https://github.com/SoftFever/BambuStudio-SoftFever/files/10426971/macros.txt)
[Test_Gcode_only_Print_End.txt](https://github.com/SoftFever/BambuStudio-SoftFever/files/10426996/Test_Gcode_only_Print_End.txt)
[Test_Gcode_M117 End Print & Print_End.txt](https://github.com/SoftFever/BambuStudio-SoftFever/files/10426997/Test_Gcode_M117.End.Print.Print_End.txt)

**Reply by erebus04** [reply]
Hi, in Your Macro File, the End Macro have Named END_PRINT.

This are the right gcode to put in the Slicer. Look at my Screenshot above.

![image](https://user-images.githubusercontent.com/34960970/212714973-ce47cfbf-e840-4e51-b47f-6dddd21bd01a.jpeg)

**Comment by 3D-Frank** [comment]
Thanks for your enlightenment!
A small mistake with a big effect

**Reply by erebus04** [reply]
No Problem,
You are welcome
